lib/typescript/GlobalButterConfig.ts

```diff
@@ -0,0 +1,9 @@
+// rome-ignore lint/suspicious/noExplicitAny: <explanation>
+type Hook = (url: string, params: Record<string, any> | undefined, responseJson?: JSON) => Promise<unknown>;
```

Comment:
You maybe consider this type being exported. It could be the case that anyone wanting to use hooks may define them outside of the buttercms call. If the type is exported, they can import the `Hook` type to ensure their callback matches what is expected.